plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEngineServerLaunchConfigurationDelegate.java

```diff
@@ -66,6 +82,12 @@ public void launch(ILaunchConfiguration configuration, String mode, ILaunch laun
 
     setDefaultSourceLocator(launch, configuration);
 
+    String pageLocation = determinePageLocation(server, configuration);
+    if (pageLocation != null) {
+      // odd: addServerListener(..., IServer.SERVER_STARTED) doesn't work
```

Comment:
Still doesn't help.  Tried `ServerEvent.STATE_CHANGE` and `ServerEvent.SERVER_CHANGE` too.